packages/router/src/page-loader.tsx

```diff
@@ -137,16 +145,23 @@ export class PageLoader extends React.Component<Props> {
     // Remove the timeout because the page has loaded.
     this.clearLoadingTimeout()
 
-    this.setState({
-      pageName: name,
-      Page: module.default,
-      slowModuleImport: false,
-      params: props.params,
+    // Update downloaded page in the Router Context to avoid
+    // blank page (page: undefined) when Route unmounts (and therefore PageLoader)
+
+    // Batched update is to avoid unnecessary re-rendering
+    unstable_batchedUpdates(() => {
+      this.props.routerSetState({ currentPage: module.default })
```

Comment:
Router context is only holding `Page`, it could be out of sync with other properties -> `params`, `pageName`.

Reply:
Moved most related state stuff to RouterContext.